### PR TITLE
[stdlib] Remove `__and__`/`__or__`/... from `Optional` and `OptionalReg`

### PR DESCRIPTION
**Pull request by soraros**
We now have guaranteed implicit conversion from `Boolable` to `Bool` which renders them useless.

### CODE REVIEW

File: stdlib/test/collections/test_optional.mojo [hunk 1]
```diff
@@ -39,10 +39,6 @@ def test_basic():
 
     assert_equal(1, a.value()[])
 
-    # Test invert operator
-    assert_false(~a)
-    assert_true(~b)
-
```
Comment (laszlokindrat) [inline stdlib/test/collections/test_optional.mojo:45]: I think Bool is getting an __invert__ method soon in another patch, can you rebase once that lands and keep these?

Comment (laszlokindrat) [reply stdlib/test/collections/test_optional.mojo:45]: nevermind, I will do this internally when I land this

**Review by laszlokindrat — APPROVED** [review]
Thanks!

File: stdlib/test/collections/test_optional.mojo (hunk 1 above)
Comment (soraros) [reply stdlib/test/collections/test_optional.mojo:45]: Ah, sorry I forgot that.

Comment (laszlokindrat) [reply stdlib/test/collections/test_optional.mojo:45]: I tried this but it actually doesn't work with implicit conversion to Bool. I'm not sure why, but I will just leave in the invert method for now.

**Comment by JoeLoser** [comment]
Landed in today's nightly: https://github.com/modularml/mojo/pull/2605.  Thanks for the contribution! 🎉

**Pull request closed without merging**